Tests/SwiftDriverTests/ModuleDependencyGraphTests.swift

```diff
@@ -1200,9 +1212,30 @@ fileprivate struct SourceFileDependencyGraphMocker {
   }
 
   private mutating func addArc(def: Node, use: Node) {
-    var use = getNode(use.sequenceNumber)
-    use.addDefIDependUpon(def.sequenceNumber)
-    allNodes[use.sequenceNumber] = use
+    while dependencyAccumulator.count <= use.sequenceNumber {
+      dependencyAccumulator.append(nil)
+    }
+    var dh = dependencyAccumulator[use.sequenceNumber] ?? {
+      let newOne = DependencyHolder()
+      dependencyAccumulator[use.sequenceNumber] = newOne
+      return newOne
+    }()
```

Comment:
I wrote this thinking this was a dictionary at first, but it's not, so this doesn't really apply. But maybe could make something like this. I think it increases readability, but completely up to you. Might be weird for an array to have that subscript.
```suggestion
    var dh = dependencyAccumulator[use.sequenceNumber, default: DependencyHolder()]
```

Related, any reason or advantage to having this be an array of optionals? Or would in make sense to append DependencyHolders in the loop instead of nil

Reply:
It is an optimization, since a number have no entries. But its a good question.